keps/sig-scheduling/4328-node-affinity-based-eviction/README.md

```diff
@@ -0,0 +1,506 @@
+- Define the behavior, syntax, and use cases in documentation.
+- Facilitate testing, adoption, and maintenance of this feature.
+
+### User Stories (Optional)
+
+#### manage multi tenant cluster with dedicated nodes
+
+I have some nodes labeled with "userA=allow" and "userB=allow", which means that these nodes are available to user A and user B.
+As the scale grows I want to add new nodes only for user B while leaving the current nodes only for user A.
+I want to migrate all pods available to user B to new nodes, and I want to ensure HA of my services during the migration.
+
+Without `node-affinity-eviction`, I have to remove "userB=allow" label of node, and delete the pods manually.
+With `node-affinity-eviction`, I can simply delete the "userB=allow" label from the existing nodes 
+to re-schedule all pods of user B to these new nodes.
+
+#### handle data locality pods when data is corrupted
+
+I have one application whose pods need to be co-located with some specific data stored in some particular nodes. 
+When one node's storage is corrupted, the pod on this node will provide some inaccurate services.
+So I want it to be scheduled to another special node when data is detected to be corrupted.
+And in the meantime, I want to keep other running pods on that node intact.
+Also, I want to ensure my PDB is guaranteed, so I can't taint the node directly.
+
+[stack overflow link](https://stackoverflow.com/questions/55006205/how-to-prevent-pods-scheduling-to-a-node-after-a-pod-has-failed-on-it-in-gce)
+
+Without `node-affinity-eviction`, I have to relabel the node to suggest data on that node is corrupted, 
+and then manually iterating the pods on the node, decide whether to delete them to trigger a reschedule.
+With `node-affinity-eviction`, the only thing I need to do is to relabel the node, and `node-affinity-eviction` controller will do the rest.
+
+### Notes/Constraints/Caveats (Optional)
+
+N/A
+
+### Risks and Mitigations
+
+The major concern may be that during the eviction, some pods may fail to be evicted. Users can use `pod_failed_eviction_node_affinity_total` metric to find them.  
+
+Typically, the reason for a failed eviction is due to a violation of the PDB. 
+However, it is also possible that there could be issues with Kubernetes components or a misconfiguration. 
+If the eviction of the target pod fails, users should first check if the eviction violates the pod's PDB configuration. 
+If it does violate the PDB, users should manually adjust the replica setting of the owner of the pod to ensure that the eviction no longer violates the PDB configuration. Alternatively, users can manually terminate the pod.
+Another thing users can check is whether there is incorrect configuration, such as multiple PodDisruptionBudgets referencing the same Pod.
+kube-apiserver and controller-manager logs can be also checked for failures.
+
+
+Another concern may be the overhead of the nodes with a large number of pods. 
```

Comment:
This concern doesn't seem significant compared to existing kubernetes features.

Reply:
removed.